src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -223,13 +221,17 @@ public static void setJDBCLoggingDirectory() {
    *
    * @param config connector configuration
    */
+  // TODO @rcheng: question - can i change this exception to also return the invalid configs?
+  //        is there a reason we want the functionality of throwing immediately here instead of collecting params?
+  //        this method is only ever called in a try catch
```

Comment:
yes, that would work! for every invalid config, only until everything is collected we throw an exception. So may be just collecting every config key value pair sounds like a good idea.

Reply:
I decided against keeping the value of the invalid config because I was worried about security (in terms of injection?). But if that is not a concern, I'd be happy to put the invalid config value in the error message along with the invalid key